src/bvals/bvals.hpp

```diff
@@ -92,6 +92,62 @@ class BoundaryBase {
   static bool called_;
 };
 
+//----------------------------------------------------------------------------------------
+//! \class BoundarySwarms
+//  \brief centralized class for interacting with each individual swarm boundary data
+class BoundarySwarms : public BoundaryBase, BoundaryCommunication {
+ public:
+  BoundarySwarms(std::weak_ptr<MeshBlock> pmb, BoundaryFlag *input_bcs,
+                 ParameterInput *pin);
+
```

Comment:
I know you are following what was done before, but I believe it is typically better practice to put different classes in separate header files, (You already did it for the sources). The reason for this is that if you make a change to your class the compiler won't have to compile the other child classes as well.

Reply:
Yeah in general I like to have each class in a separate header file, if only for stylistic reasons. In this case I left it in the same file because I wanted to maintain the analogy to `BoundaryValues`, for the benefit of anyone who tries to reorganize the boundaries code. If you'd rather I make a new file for this class though I'm fine with doing that.